tests/test_vectorized.py

```diff
@@ -325,133 +322,8 @@ def test_enormous_through_ares():
-
-
-def test_broadcast_customtransfermap():
-    """Test that broadcasting a `CustomTransferMap` element gives the correct result."""
-    tm = torch.tensor(
-        [
-            [
-                [1.0, 4.0e-02, 0.0, 0.0, 0.0, 0.0, 0.0],
-                [0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0e-05],
-                [0.0, 0.0, 1.0, 4.0e-02, 0.0, 0.0, 0.0],
-                [0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0],
-                [0.0, 0.0, 0.0, 0.0, 1.0, -4.6422e-07, 0.0],
-                [0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0],
-                [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0],
-            ]
-        ]
-    )
-
-    element = cheetah.CustomTransferMap(length=torch.tensor([0.4]), transfer_map=tm)
-    broadcast_element = element.broadcast((3, 10))
-
-    assert broadcast_element.length.shape == (3, 10)
-    assert broadcast_element._transfer_map.shape == (3, 10, 7, 7)
-    for i in range(3):
-        for j in range(10):
-            assert torch.all(broadcast_element._transfer_map[i, j] == tm[0])
-
-
-def test_broadcast_element_keeps_dtype():
-    """Test that broadcasting an element keeps the same dtype."""
-    element = cheetah.Drift(length=torch.tensor([0.4]), dtype=torch.float64)
-    broadcast_element = element.broadcast((3, 10))
-
-    assert broadcast_element.length.dtype == torch.float64
-
-
-def test_broadcast_beam_keeps_dtype():
-    """Test that broadcasting a beam keeps the same dtype."""
-    beam = cheetah.ParticleBeam.from_parameters(
-        num_particles=100_000, sigma_x=torch.tensor([1e-5]), dtype=torch.float64
-    )
-    broadcast_beam = beam.broadcast((2,))
-    drift = cheetah.Drift(length=torch.tensor([0.4, 0.4]), dtype=torch.float64)
-
-    assert broadcast_beam.particles.dtype == torch.float64
-
-    # This should not raise an error
-    _ = drift(broadcast_beam)
-
-
-def test_broadcast_drift():
-    """Test that broadcasting a `Drift` element gives the correct result."""
-    element = cheetah.Drift(length=torch.tensor([0.4]))
-    broadcast_element = element.broadcast((3, 10))
-
-    assert broadcast_element.length.shape == (3, 10)
-    for i in range(3):
-        for j in range(10):
-            assert broadcast_element.length[i, j] == 0.4
-
-
-def test_broadcast_quadrupole():
-    """Test that broadcasting a `Quadrupole` element gives the correct result."""
-
-    # TODO Add misalignment to the test
-    # TODO Add tilt to the test
-
-    element = cheetah.Quadrupole(length=torch.tensor([0.4]), k1=torch.tensor([4.2]))
-    broadcast_element = element.broadcast((3, 10))
-
-    assert broadcast_element.length.shape == (3, 10)
-    assert broadcast_element.k1.shape == (3, 10)
-    for i in range(3):
-        for j in range(10):
-            assert broadcast_element.length[i, j] == 0.4
-            assert broadcast_element.k1[i, j] == 4.2
+    assert outgoing.energy.shape == torch.Size([1])
+    assert outgoing.total_charge.shape == torch.Size([1])
```

Comment:
```suggestion
    assert outgoing.energy.shape == (3, 100_000)
    assert outgoing.total_charge.shape == (3, 100_000)
```
I think these should actually also come out in the broadcast shape. Image running a `Cavity` at different settings. You would get different beams with different energies. So every `Element` should give you different energies for different settings, even if they end up being the same value.

The reason the broadcast fails here is because most elements will just copy over the energy. So we probably need a broadcasting alternative to that or just do it by hand.